### PR TITLE
Move new platform support guide to k/sig-release

### PR DESCRIPTION
**Pull request by hasheddan**
**Which issue(s) this PR fixes**:

Moves new platform support guide to live alongside documentation of
support tiers and tier 3 build docs.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

xref https://github.com/kubernetes/sig-release/pull/1451 https://github.com/kubernetes/sig-release/pull/1444#discussion_r567498462 https://github.com/kubernetes/sig-release/issues/1337

/assign @dims

### CODE REVIEW

**Comment by dims** [comment]
/approve

please lgtm after the other pr merges ( https://github.com/kubernetes/sig-release/pull/1451 )

**Review by saschagrunert — APPROVED** [review]
> /approve
> 
> please lgtm after the other pr merges ( [kubernetes/sig-release#1451](https://github.com/kubernetes/sig-release/pull/1451) )

/lgtm

**Pull request merged**
